src/main/java/pl/edu/zut/mad/schedule/search/SearchInputFragment.kt

```diff
@@ -0,0 +1,113 @@
+package pl.edu.zut.mad.schedule.search
+
+import android.app.DatePickerDialog
+import android.os.Bundle
+import android.support.annotation.StringRes
+import android.support.design.widget.Snackbar
+import android.support.v4.app.Fragment
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import android.widget.EditText
+import android.widget.Toast
+import kotlinx.android.synthetic.main.fragment_search_input.*
+import org.joda.time.LocalDate
+import pl.edu.zut.mad.schedule.R
+import pl.edu.zut.mad.schedule.ScheduleDate
+import pl.edu.zut.mad.schedule.data.model.ui.Lesson
+import pl.edu.zut.mad.schedule.util.app
+import javax.inject.Inject
+
+internal class SearchInputFragment : Fragment(), SearchMvp.View {
+
+    companion object {
+        private const val DAYS_IN_WEEK = 7
+    }
+
+    @Inject internal lateinit var presenter: SearchMvp.Presenter
+
+    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View =
+        inflater.inflate(R.layout.fragment_search_input, container, false)
+
+    override fun onViewCreated(view: View?, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+        init()
+    }
+
+    override fun getTeacherName() = teacherNameInputView.text.toString()
```

Comment:
Maybe add some view model/s?

Reply:
In next PR I will change ui.Lesson model, so multiples getters will replaced.